src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +265,41 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
-                && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                && !isEmptyBody(details.lcurly);
+    }
+
+    /**
+     * Checks that a right curly should be alone on a line or allows single-line format of block.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on a line or single-line format of block.
+     */
+    private static boolean shouldBeAloneOrSLine(RightCurlyOption bracePolicy, Details details) {
```

Comment:
@romani Shouldn't `AbbreviationAsWordInName` have picked it up?
It seems [the check must be set to 0](https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L348) to pick this type of name up.